File: Backend/Models/Unidade.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    [Table("Unidade")]
    public class Unidade
    {
        [Key]
        public int Id { get; set; }

        
        public string NomeUnidade { get; set; }

        [ForeignKey("Empresa")]
        public int IdEmpresa { get; set; }
        public virtual Empresa? Empresa { get; set; }

        [ForeignKey("Endereco")]
        public int IdEndereco { get; set; }
        public virtual Endereco? Endereco { get; set; }
    }
}

File: Frontend/src/app/components/interface/unidade.ts
import { Endereco } from './endereco';
import { Empresa } from './empresa';

export interface Unidade {
    Id: number;
    NomeUnidade: string;
    IdEmpresa: number;
    IdEndereco: number;
    Endereco?: Endereco;
    Empresa?: Empresa;
}